From 8bca02efcee09422c8cc14bccb7e19388dfe74bb Mon Sep 17 00:00:00 2001
Message-Id: <patch-2.2-8bca02efcee-20210505T121028Z-avarab@gmail.com>
In-Reply-To: <cover-0.2-00000000000-20210505T121028Z-avarab@gmail.com>
References: <cover-0.2-00000000000-20210505T121028Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Wed, 5 May 2021 12:58:22 +0200
Subject: [PATCH 2/2] sparse-index.c: don't call prepare_repo_settings() twice
 in a row
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change code added in 58300f47432 (sparse-index: add index.sparse
config option, 2021-03-30) to only call prepare_repo_settings()
once. We know that our own set_sparse_index_config() has just finished
calling it, so we don't need to call it if we're acting on "test_env".

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 sparse-index.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sparse-index.c b/sparse-index.c
index b0d5dc5f081..5bad05de645 100644
--- a/sparse-index.c
+++ b/sparse-index.c
@@ -133,11 +133,12 @@ int convert_to_sparse(struct index_state *istate)
 	test_env = git_env_bool("GIT_TEST_SPARSE_INDEX", -1);
 	if (test_env >= 0)
 		set_sparse_index_config(istate->repo, test_env);
+	else
+		prepare_repo_settings(istate->repo);
 
 	/*
 	 * Only convert to sparse if index.sparse is set.
 	 */
-	prepare_repo_settings(istate->repo);
 	if (!istate->repo->settings.sparse_index)
 		return 0;
 
-- 
2.31.1.838.g7ac6e98bb53

